internal/deviations/deviations.go

```diff
@@ -66,9 +66,34 @@ package deviations
 import (
 	"flag"
 
+	log "github.com/golang/glog"
+	"github.com/openconfig/featureprofiles/internal/metadata"
+	mpb "github.com/openconfig/featureprofiles/proto/metadata_go_proto"
 	"github.com/openconfig/ondatra"
 )
 
+// lookupDutDeviations returns the deviations for the specified dut. If no duts match `nil` is returned.
+func lookupDUTDeviations(dut *ondatra.DUTDevice) *mpb.Metadata_Deviations {
+	for _, platformExceptions := range metadata.Get().PlatformExceptions {
+		if platformExceptions.GetPlatform().Vendor.String() == dut.Device.Vendor().String() && platformExceptions.GetPlatform().HardwareModel == dut.Device.Model() {
+			return platformExceptions.GetDeviations()
+		}
+	}
+	log.Warningf("No platform exceptions for dut %v configured in platform exceptions metadata %v", dut, metadata.Get().PlatformExceptions)
+	return nil
+}
+
+// isFlagSetByUser returns if the specified flag value is set.
+func isFlagSetByUser(name string) bool {
+	found := false
+	flag.VisitAll(func(f *flag.Flag) {
```

Comment:
I think you need Visit, not VisitAll? Visit only visits flags that are set by the user

Reply:
done